xm-commons-data-spec/src/main/java/com/icthh/xm/commons/service/DefaultSpecProcessingService.java

```diff
@@ -40,8 +45,18 @@ public <I extends SpecificationItem> void processDataSpecification(String tenant
     }
 
     @Override
-    public void updateByTenantState(String tenant, String baseSpecKey, S spec) {
-        definitionSpecProcessor.updateStateByTenant(tenant, baseSpecKey, spec.getDefinitions());
-        formSpecProcessor.updateStateByTenant(tenant, baseSpecKey, spec.getForms());
+    public void fullUpdateByTenantState(String tenant, String baseSpecKey, List<S> specs) {
+        final List<DefinitionSpec> allSpecDefinitions = specs.stream()
+            .filter(spec -> Objects.nonNull(spec) && Objects.nonNull(spec.getDefinitions()))
+            .flatMap(spec -> spec.getDefinitions().stream())
+            .toList();
+
+        final List<FormSpec> allSpecForms = specs.stream()
+            .filter(spec -> Objects.nonNull(spec) && Objects.nonNull(spec.getForms()))
+            .flatMap(spec -> spec.getForms().stream())
+            .toList();
```

Comment:
seems like a redundant iterations of quite an amount of instances since this validations are present in future updateStateByTenant logic

when have you met an exception to add this iterations? is it possible to make it safe in place the error occurred to avoid it happening if the dev forgot to do this validation?